src/moduleScript.js

```diff
@@ -1,33 +1,59 @@
 import { Module } from "module";
+import { AstSerializer } from "./ast.js";
 
 class ModuleScript {
 
 	static CJS_MODULE_EXPORTS = "module.exports = ";
 	static ESM_EXPORT_DEFAULT = "export default ";
 	static FUNCTION_REGEX = /^(?:async )?function\s?\S*\(/;
 
+	static ASYNC_CONSTRUCTOR = (async function () {}).constructor;
+	static SYNC_CONSTRUCTOR = (function () {}).constructor;
+
 	static getProxiedContext(context, propertyReferenceKey, propertyValue) {
 		let proxiedContext = new Proxy(context, {
 			get(target, propertyName) {
 				if(Reflect.has(target, propertyName)) {
 					return Reflect.get(target, propertyName);
 				}
-				throw new Error(`'${propertyName}' not found when evalutating ${propertyReferenceKey} with value '${propertyValue}'.
-Check that '${propertyName}' is a valid attribute or property name, is present in global data, or is a helper.`);
+				// TODO show file name!!
+				throw new Error(`'${propertyName}' not found when evaluating ${propertyReferenceKey}="${propertyValue}".
+Check that '${propertyName}' is a helper, attribute name, property name, or is present in global data.`);
 			}
 		});
 
 		return proxiedContext;
 	}
 
-	static evaluateAsyncAttribute(content) {
-		const AsyncFunction = (async function () {}).constructor;
-		return new AsyncFunction(`return ${content};`);
+	static addDestructuredWith(content, data = {}) {
+		let keys = Object.keys(data).map(attr => attr.startsWith(AstSerializer.prefixes.dynamic) ? attr.slice(1) : attr);
+		return `return ({${keys.join(",")}}) => ${content};`;
+	}
```

Comment:
**Issue**: This will create invalid destructuring expressions for objects with number keys like this:

```js
let data = { 3: 'value' }
let { 3 } = data // SyntaxError: Unexpected number
```

So when later, the following call is executed:

```js
new Function(`return ({3}) => 'value';`) // SyntaxError: Unexpected token '}'
```

The number key case is the only one of the ones I added in #35 that fails in your implementation.